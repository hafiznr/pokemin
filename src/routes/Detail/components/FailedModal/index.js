/** @jsxRuntime classic */
/** @jsx jsx */
import { 
  Button, 
  Modal, 
  ModalBody, 
  ModalContent, 
  ModalHeader, 
  ModalOverlay 
} from "@chakra-ui/react";
import { jsx, css } from "@emotion/react";

const FailedModal = ({
  isOpen,
  pokemon = '',
  image,
  onClose,
  onRetry
}) => {
  const center = css({
    textAlign: 'center'
  });

  const imgStyle = css({
    display: 'inline'
  });

  return (
    <Modal isOpen={isOpen} onClose={onClose} size="xs" isCentered>
      <ModalOverlay  />
      
      <ModalContent>
        <ModalHeader>
          <div css={center}>Aargh! Almost had it!</div>
        </ModalHeader>
        <ModalBody>
          <div css={center}>
            <img 
              src={image} 
              alt={pokemon} 
              width="96px" 
              height="96px" 
              css={imgStyle}
              loading="lazy" 
            />
            <br />
            <Button
              colorScheme="twitter"
              size="lg"
              width="100%"
              onClick={onRetry}
            >
              Try again!
            </Button>
            <Button
              m="8px 0"
              size="lg"
              width="100%"
              onClick={onClose}
            >
              Close
            </Button>
          </div>
        </ModalBody>
      </ModalContent>
    </Modal>
  )
}

export default FailedModal;