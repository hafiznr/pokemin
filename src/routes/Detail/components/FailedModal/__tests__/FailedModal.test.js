import { render } from '@testing-library/react';
import FailedModal from '../';

test('renders FailedModal', () => {
  const { container } = render(<FailedModal/>);

  expect(container).toMatchSnapshot();
});