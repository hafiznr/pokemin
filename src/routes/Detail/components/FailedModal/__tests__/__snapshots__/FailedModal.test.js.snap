// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FailedModal 1`] = `<div />`;
